lib/problem3.rb

```diff
@@ -0,0 +1,12 @@
+def problem3(values)
+  incremental_product = 1
+  index_counter = 1
+  array_product = 1
+  while value = values.shift do
+    array_product *= value
+    incremental_product *= index_counter
```

Comment:
You can get rid of this statements and related ones to it. As you receive an array, you have the size of it, so you may get the factorial of it. See: https://stackoverflow.com/questions/2434503/ruby-factorial-function